### PR TITLE
Convert topology user manual to ReStructuredText

### PR DESCRIPTION
**Pull request by terhorstd**
@jessica-mitchell worked on the conversion towards [this branch](https://github.com/terhorstd/nest-simulator/pull/4):

> ... in preparation to build an acceptable markdown version to be used as the default topology user manual.
> The fixes resolve several outstanding issues:  (note the commit messages have some details about changes to the patch and Snakefile.) 
> 1) Python code snippets generate properly in html/md/rst and not just pdf
> 2) Figures, Table and equations are numbered and have linked references (to generate these I installed pandoc-fignos / eqnos / tablenos - see https://github.com/tomduck)
> 3) Missing pngs now exist and are properly referenced 
> 4) Bibliography is now generated, with a 'new' csl format (the format can be further discussed) and in-text citations link to bibliography
> 5) Sections now appear numbered in pdf and html
> 6) Tables appear decent - in html, pdf (need further work in rst)
> 
> TODO
> - clean up the Snakefile - the pandoc lines have become much longer and some of this may be can be put into metadata.yaml (including the bib.yaml) or condensed in other ways
> - no table of figures / tables  in pdf/html format were added (do we need this?)
> - figure size may need to be altered 
> - tables and equations should probably be checked over 
> - check for any other mistakes!
> 
fixes [#374](https://github.com/nest/nest-simulator/issue/374)

### CODE REVIEW

**Comment by gtrensch** [comment]
The html-version of the Topology user manual looks in general very good to me! I found some formatting issues that need to be fixed:

- formula not rendered
-----------------------------
image

- code example formatting
---------------------------------
image

- text background should not be grey
----------------------------------------------
image

- formatting of the table: it is not fully clear which parameter belongs to which name
---------------------------------------------------------------------------------------------------------
image

- example code is missing: formatting/rendering needs to be fixed
-----------------------------------------------------------------------------------
image

- example code is missing: formatting/rendering needs to be fixed
-----------------------------------------------------------------------------------
image

- example code is missing: formatting/rendering needs to be fixed
-----------------------------------------------------------------------------------
image

- example code is missing: formatting/rendering needs to be fixed
-----------------------------------------------------------------------------------
image

**Comment by Silmathoron** [comment]
Hey there, so what is the status of this PR with respect to 2.18? Do you still want to convert it to rst? Is help still needed regarding the [inclusion of code](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#include-code-with-the-literalinclude-directive)?

**Comment by steffengraber** [comment]
Tested everything and it worked  - nice!

File: doc/README.rst [hunk 1]
```diff
@@ -6,6 +6,37 @@ on improving the documentation and making updates. However, if you notice
...
+-----------------------
+
+You will need the NEST source code in your project::
+
+
+    cd nest-simulator/doc/
+
+Requirements for building the docs include::
+
+ sphinx
```
Comment (gtrensch) [inline doc/README.rst]:
```suggestion
 python 3.6
 setuptools
 pandoc
 sphinx
```

File: doc/README.rst [hunk 2]
```diff
@@ -6,6 +6,37 @@ on improving the documentation and making updates. However, if you notice
...
+ recommonmark
+ sphinx-rtd-theme
+ nbsphinx
+ breathe
+ sphinx-gallery
+ ipython
+ mock
+
+
+You can now build the docs using this commandi (make sure you are in the doc folder)::
```
Comment (gtrensch) [inline doc/README.rst]:
```suggestion
You can now build the docs using this command (make sure you are in the doc folder)::
```

**Pull request merged**